model/messages/dkg.go

```diff
@@ -37,8 +35,22 @@ type PrivDKGMessageOut struct {
 
 // BroadcastDKGMessage is a wrapper around a DKGMessage intended for broadcasting.
 // It contains a signature of the DKGMessage signed with the staking key of the
-// sender.
+// sender. DTTF that BroadcastDKGMessage are communicated via the DKG smart contract
+// we must also include the Orig or DKG committee index of the sender so that when these
+// messages are consumed we can verify if BroadcastDKGMessage was sent by a DKG committee member.
```

Comment:
We can allow the whiteboard to tell us the sender of a broadcast message, in the same way the network layer does for private message, which may simplify things here. 

The first field of the value we read from the smart contract https://github.com/onflow/flow-go/blob/d75191e2b0077c47fe5c46678baeca1df5cd0bb9/module/dkg/client.go#L141-L145 is the node ID of the message sender. (eg. `nodeID := val.(cadence.Struct).Fields[0]`

Reply:
https://github.com/onflow/flow-go/pull/2238/commits/578700dc80f04d5ffc1c75c4975b31e2f159f81e

I added json ignore tags so that Orig and NodeID aren't sent when broadcasting messages, instead I am using the nodeID returned from the contract to later set the Orig field when processing the broadcast messages during reading

Reply:
I agree we do not need to add `nodeID` or `orig` as fields. However, how can we read the whiteboard data when printed in a file?
Debugging is still important at this stage. If I could keep one of the 2 fields, I would prefer to keep `orig`. It's only a few characters more per message.

Reply:
I agree we do not need to add `nodeID` or `orig` as fields. However, how can we read the whiteboard data when printed in a file?
Debugging is still important at this stage. If I could keep one of the 2 fields, I would prefer to keep `orig`. It's only a few characters more per message.